testing/trino-tests/src/test/java/io/trino/execution/TestEventListenerBasic.java

```diff
@@ -386,45 +426,75 @@ public void testReferencedTablesWithViews()
         assertThat(tables).hasSize(2);
 
         TableInfo table = tables.get(0);
-        assertTableInfoBasics(table, "tpch", "tiny", "nation", "user", false, 1, 0);
+        assertTableInfoBasics(table, "tpch", "tiny", "nation", "user", false, 1, 0, 1);
```

Comment:
As a followup we could use a fluent type of APIs to perform the assertion. (No changes requested - for my reference)

Reply:
Yah had this thought too actually, but didn't feel ready to dive into that ...

Reply:
Actually, you inspired me :) Spent a little time to try my hand at making fluent-style assertions. A fun exercise! LMK what you think.

I left the old branch w/o fluent assertions available at: https://github.com/xkrogen/trino/tree/xkrogen/genericreferencechain-no-fluent-assertions